systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,214 @@
+     * │ - Threads are managed by an executor service that efficiently schedules each batch               │
+     * │   task to maximize the available CPU resources.                                                  │
+     * │ - Errors in any batch operation are logged and do not impact the processing of other             │
+     * │   batches, ensuring robust execution.                                                            │
+     * └──────────────────────────────────────────────────────────────────────────────────────────────────┘
+     *
+     * @param testStorage    The storage instance containing the configuration and namespace details.
+     * @param numberOfTopics The total number of Kafka topics to process.
+     */
+    public static void processKafkaTopicBatchesAsync(TestStorage testStorage, int numberOfTopics) {
+        final int availableCPUs = Runtime.getRuntime().availableProcessors();
+        final int optimalBatchCount = Math.max(1, availableCPUs);
+        final int topicsPerBatch = (numberOfTopics + optimalBatchCount - 1) / optimalBatchCount; // Ensures all topics are covered
+
+        final ExecutorService executor = Executors.newFixedThreadPool(optimalBatchCount);
+        final ExtensionContext currentContext = ResourceManager.getTestContext();
+        final CompletableFuture<?>[] futures = new CompletableFuture[optimalBatchCount];
+
+        try {
+            for (int batch = 0; batch < optimalBatchCount; batch++) {
+                int start = batch * topicsPerBatch;
+                int end = Math.min(start + topicsPerBatch, numberOfTopics); // Ensure we do not go out of bounds
+                // Delegate batch processing to a separate method
+                futures[batch] = processBatch(start, end, currentContext, testStorage, executor);
+            }
+            CompletableFuture.allOf(futures).join(); // Wait for all batches to complete
+            LOGGER.info("All batches completed.");
+        } finally {
+            executor.shutdown();
+        }
+    }
+
+    /**
+     * Manages the creation, modification, and deletion of Kafka topics for a specific batch.
+     * Operations are performed sequentially within each batch but executed concurrently across multiple batches,
+     * utilizing multiple threads to improve performance.
+     *
+
+     *
+     * @param start           Starting index of the batch.
+     * @param end             Ending index of the batch.
+     * @param currentContext  The current JUnit extension context.
+     * @param testStorage     Storage instance with namespace and configuration details.
+     * @param executor        The executor service managing concurrent tasks.
+     * @return                CompletableFuture representing the completion of all operations in the batch.
+     */
+    private static CompletableFuture<Void> processBatch(int start, int end, ExtensionContext currentContext,
```

Comment:
The way you process the batch is not similar to what a normal user would do. He would create 1 topic, do some updates, and maybe delete it when done. Instead here we are creating, updating and deleting all at once.

Reply:
Sure I have impl. such approach in next commit :) check it